### PR TITLE
Added channel aware redirects

### LINKED ISSUE #3: Enable redirects on specific channels
Allow the user to only enable a redirect for certain channels

### PR DESCRIPTION
**Pull request by loevgaard**
Fixes #3

### CODE REVIEW

File: src/Repository/RedirectRepository.php [hunk 1]
```diff
@@ -7,45 +7,98 @@
...
+                ->leftJoin('r.channels', 'c')
+            ;
+        }
+
         return $qb->getQuery()
             ->getOneOrNullResult();
     }
 
-    public function removeNotAccessed(int $threshold): void
+    public function searchNextRedirectByChannel(RedirectInterface $redirect, ChannelInterface $channel, bool $only404 = false): ?RedirectInterface
```
Comment (igormukhingmailcom) [inline src/Repository/RedirectRepository.php]: Looks like we can omit this function. Or it required for BC?

File: src/Resources/config/doctrine/model/Redirect.orm.xml [hunk 2]
```diff
@@ -9,10 +9,6 @@
             <generator strategy="AUTO"/>
         </id>
 
-        <unique-constraints>
-            <unique-constraint columns="source"/>
```
Comment (igormukhingmailcom) [inline src/Resources/config/doctrine/model/Redirect.orm.xml]: Should probably be `source, channel_id`

File: src/Repository/RedirectRepository.php (hunk 1 above)
Comment (loevgaard) [reply src/Repository/RedirectRepository.php]: Yes, we can't remove any methods :(

File: src/Resources/config/doctrine/model/Redirect.orm.xml (hunk 2 above)
Comment (loevgaard) [reply src/Resources/config/doctrine/model/Redirect.orm.xml]: Yes, but is that possible? It's a many to many relationship

**Pull request merged**